pkg/issuer/acme/dns/rfc2136/tsig_test.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2020 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package rfc2136 implements a DNS provider for solving the DNS-01 challenge
+// using the rfc2136 dynamic update.
+
+package rfc2136
+
+import (
+	"crypto/hmac"
+	"crypto/md5"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/base64"
+	"reflect"
+	"testing"
+
+	"github.com/miekg/dns"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_tsigHMACProvider_Generate(t *testing.T) {
```

Comment:
Note: this unit test is largely just testing that the upstream libaries work as expected, but would catch a very basic error i.e in the switch and serves as a very basic documentation for what the Generate method does.

Reply:
Yeah, I wasn't sure it was necessary. But I did check the coverage which seems fine.

![image](https://user-images.githubusercontent.com/978965/159032491-886b1a5a-ffbc-47c5-838c-bcbdddddb2a0.png)